### ISSUE TITLE
Add ability to launch into cgroups

### ISSUE THREAD

**Issue by a10y** [palantir/go-java-launcher#128]
Feature Request: add ability to launch into cgroups.

Internally we have a number of services that call into native libraries from Java. When either the FFI code or the actual native code has bugs, this can occasionally lead to memory leaks which don't surface at the JVM level as OutOfMemoryErrors, but instead continue to cause process RSS to rise until it eventually consumes all physical memory in the system.

An easy way to prevent this from happening is to place the process into a cgroup (which is what docker/other container runtimes do behind the scenes) and set fine-grained memory limits on a process.

Adding a process into a cgroup is really easy: you just write a PID into a pseudo-file, e.g.

```
$ top &
$ echo $! > /sys/fs/cgroup/memory/my-group/tasks
```

I'd like to add the ability to allow go-java-launcher to launch a process into a pre-created cgroup, which can be configured via the `launcher-custom.yml` file.

**Comment by a10y** [comment]
Would likely only support cgroupsv1 as a first pass, and likely only memory controllers. pseudo-conjure syntax:

```yaml
CgroupConfig:
  enum:
    - CgroupV1Config
CgroupV1Config:
  fields:
    memoryGroup: string
```

**Comment by a10y** [comment]
Merged in #129
